packages/blocks/src/block-content-provider/index.js

```diff
@@ -23,28 +30,43 @@ import { serialize } from '../api';
  * </BlockContentProvider>
  * ```
  */
-class BlockContentProvider extends Component {
-	getChildContext() {
-		const { innerBlocks } = this.props;
+const BlockContentProvider = ( { children, innerBlocks } ) => {
+	const BlockContent = () => {
+		// Value is an array of blocks, so defer to block serializer
+		const html = serialize( innerBlocks );
 
-		return {
-			BlockContent() {
-				// Value is an array of blocks, so defer to block serializer
-				const html = serialize( innerBlocks );
+		// Use special-cased raw HTML tag to avoid default escaping
+		return <RawHTML>{ html }</RawHTML>;
+	};
 
-				// Use special-cased raw HTML tag to avoid default escaping
-				return <RawHTML>{ html }</RawHTML>;
-			},
-		};
-	}
-
-	render() {
-		return this.props.children;
-	}
-}
-
-BlockContentProvider.childContextTypes = {
-	BlockContent: () => {},
+	return (
+		<Provider value={ { BlockContent } }>
```

Comment:
We pass a new value each render. I suppose this isn't a huge issue since it's used only in the serialization step anyways, but why does the context need to be an object anyways? Why not just the `BlockContent` value directly?